docs/sources/configure-client/grafana-agent/go_pull.md

```diff
@@ -0,0 +1,238 @@
+| -------- | ------- | ------------------------------------------ | ---------------------- | -------- |
+| enabled  | boolean | Enable this profile type to be scraped.    | true                   | no       |
+| path     | string  | The path to the profile type on the target. | "/debug/pprof/profile" | no       |
+| delta    | boolean | Whether to scrape the profile as a delta.  | true                   | no       |
+
+### `profile.fgprof`
+
+Collects profiles from an fgprof endpoint.
+
+| Argument | Type    | Description                                | Default           | Required |
+| -------- | ------- | ------------------------------------------ | ----------------- | -------- |
+| enabled  | boolean | Enable this profile type to be scraped.    | false             | no       |
+| path     | string  | The path to the profile type on the target. | "/debug/fgprof"   | no       |
+| delta    | boolean | Whether to scrape the profile as a delta.  | true              | no       |
+
+### `profile.godeltaprof_memory`
+
+Collects profiles from godeltaprof memory endpoint. The delta is computed on the target.
+
+| Argument | Type    | Description                                | Default                  | Required |
+| -------- | ------- | ------------------------------------------ | ------------------------ | -------- |
+| enabled  | boolean | Enable this profile type to be scraped.    | false                    | no       |
+| path     | string  | The path to the profile type on the target. | "/debug/pprof/delta_heap" | no       |
+
+### `profile.godeltaprof_mutex`
+
+Collects profiles from godeltaprof mutex endpoint. The delta is computed on the target.
+
+| Argument | Type    | Description                                | Default                   | Required |
+| -------- | ------- | ------------------------------------------ | ------------------------- | -------- |
+| enabled  | boolean | Enable this profile type to be scraped.    | false                     | no       |
+| path     | string  | The path to the profile type on the target. | "/debug/pprof/delta_mutex" | no       |
+
+### `profile.godeltaprof_block`
+
+Collects profiles from godeltaprof block endpoint. The delta is computed on the target.
+
+| Argument | Type    | Description                                | Default                   | Required |
+| -------- | ------- | ------------------------------------------ | ------------------------- | -------- |
+| enabled  | boolean | Enable this profile type to be scraped.    | false                     | no       |
+| path     | string  | The path to the profile type on the target. | "/debug/pprof/delta_block" | no       |
+
+## Example
+
+The following configuration sets up a scraping job that scrapes two local applications (the Agent itself and Pyroscope). The profiles obtained are then sent over to the receivers as defined by other components.
+
+```river
```

Comment:
@korniltsev can you help me improve these docs -- I know we have a whole separate agent page, but I would like for people to be able to get a minimum amount of information needed to successfully send data from Grafana Agent (pull mode) to their local Pyroscope or to Grafana cloud without leaving this page. 

https://github.com/grafana/pyroscope/blob/main/examples/grafana-agent/agent/config/config.river

in this doc ^^ we have a config that looks likes this:
```
logging {
  level = "debug"
  format = "logfmt"
}

pyroscope.write "example" {
  // Send metrics to a locally running Phlare instance.
  endpoint {
    url = "http://pyroscope:4100"

    // To send data to Grafana Cloud you'll need to provide username and password.
    // basic_auth {
    //   username = "myuser"
    //   password = "mypassword"
    // }
  }
  external_labels = {
    "env" = "example",
  }
}


pyroscope.scrape "default" {
  targets = [
    {"__address__" = "pyroscope:4100", "service_name"="pyroscope"},
    {"__address__" = "agent:12345", "service_name"="agent"},
  ]
  forward_to = [pyroscope.write.example.receiver]
}
```

How do I update this code snippet to include both the scrape jobs, but also add a receiver that will successfully receive the profiles from this scrape job (in both oss and cloud) 

cc @simonswine @petethepig

Reply:
```
pyroscope.write "example" {
	endpoint {
		url = "http://pyroscope:4100"
		// To send data to Grafana Cloud you'll need to provide username and password. //
		// basic_auth {
		//	username = "myuser"
		//	password = "mypassword"
		//}
	}
}

pyroscope.scrape "default_settings" {
	targets    = [{"__address__" = "localhost:12345", "service_name" = "example_service"}]
	forward_to = [pyroscope.write.example.receiver]

	profiling_config {
		profile.goroutine {
			enabled = true
			path    = "/debug/pprof/goroutine"
			delta   = false
		}

		profile.mutex {
			enabled = true
			path    = "/debug/pprof/mutex"
			delta   = false
		}

		profile.process_cpu {
			enabled = true
			path    = "/debug/pprof/profile"
			delta   = true
		}

		profile.godeltaprof_memory {
			enabled = false
			path    = "/debug/pprof/delta_heap"
		}

		profile.memory {
			enabled = true
			path    = "/debug/pprof/memory"
			delta   = false
		}

		profile.godeltaprof_mutex {
			enabled = false
			path    = "/debug/pprof/delta_mutex"
		}

		profile.godeltaprof_block {
			enabled = false
			path    = "/debug/pprof/delta_block"
		}

		profile.block {
			enabled = true
			path    = "/debug/pprof/block"
			delta   = false
		}
	}
}

```

Reply:
profiling_config defaults are good enough by default. I would just remove the `profiling_config` block

Reply:
I think its good to list them out here because otherwise people may not know what the defaults are -- also should peopel be using `profile.godeltaprof_block ` or `profile.block` ?

Reply:
`godeltaprof_block` is better than `block` in terms of cpu&mem consumption. But for `godeltaprof_block` they need an extra dependency and import in their app.

Reply:
where is this documented (if anywhere)?